### ISSUE TITLE
Context "override" does not work for behave like statement?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [etorreborre/specs#53]
```
What steps will reproduce the problem?

Here is my simple test program:

    import org.specs._

    object BehaveLikeBugSpec extends Specification {
      val systemCtxt1 = new SystemContext[String] {
        def newSystem =  {
          println(">>>> #1 system context " )
          "abc"
        }
      }
      val systemCtxt2 = new SystemContext[String] {
        def newSystem =  {
          println(">>>> #2 system context " )
          "123"
        }
      }
      "fisrt spec" definedAs(systemCtxt1) should {
        "do dummy test" in { x: String =>
          println("----- dummy test: " + x)
        }
      }
      "second spec" definedAs(systemCtxt2) should {
        "behave like example" in { x: String =>
          behave like "fisrt spec"
        }
      }
    }


What is the expected output? What do you see instead?
I've expected that 2ns system context ("#2 system context") will be passed 
to the reused example from the first SUT on execution of the "behave like" 
statement. 
Instead, the 1st system context is still passed to the reused example.

Here is a trace:

    Specification "BehaveLikeBugSpec"
      fisrt spec should
    >>>> #1 system context 
    ----- dummy test: abc
      + do dummy test

      Total for SUT "fisrt spec":
      Finished in 0 second, 0 ms
      1 example, 0 expectation, 0 failure, 0 error

      second spec should
    >>>> #2 system context 
    >>>> #2 system context 
    >>>> #1 system context 
    ----- dummy test: abc
          + do dummy test

      Total for SUT "second spec":
      Finished in 0 second, 0 ms
      1 example, 0 expectation, 0 failure, 0 error

    Total for specification "BehaveLikeBugSpec":
    Finished in 0 second, 22 ms
    2 examples, 0 expectation, 0 failure, 0 error



What version of the product are you using? On what operating system?

1.4.3-SNAPSHOT

Please provide any additional information below.

It seems like the system context is always bound to the example instance. 
As a result, on cloning of the example, the "old" context is copied too. 
Perhaps new context should be set (if defined) after cloning?

BTW, similar behavior is exposed by stackSpec.scala example program: the 
same stack is tested regardless of the passed context.



```

Original issue reported on code.google.com by `udovenk...@gmail.com` on 25 Jan 2009 at 7:01

**Comment by GoogleCodeExporter** [comment]
Original comment by `etorrebo...@gmail.com` on 27 Jan 2009 at 1:44
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for your report, that was indeed a big hole in the "context" machinery. 
I've 
fixed it in the latest version of specs-1.4.3-SNAPSHOT.

Note also that your test case is simpler if you write the second spec as:

 "second spec" definedAs(systemCtxt2) should {
   behave like "fisrt spec"
 }

```

Original comment by `etorrebo...@gmail.com` on 29 Jan 2009 at 1:52
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `etorrebo...@gmail.com` on 25 Feb 2009 at 9:37
- Changed state: **Verified**
